execute/plugin.go

```
@@ -254,16 +253,14 @@ func selectReport(
 			return nil, nil, fmt.Errorf("unable to add report to builder: %w", err)
 		}
 
-		selectedReports = append(selectedReports, commitReports[i])
```

This line is buggy. It will include commit reports even if the result of `Add` is `ErrEmptyReport`, which leads to a mismatch between the returned `execReports` and `selectedReports` lists.

---

I don't get how it's going to include the commit report when there's an error? we have an error check before this. and in my other PR I'm doing continue which will also prevent it from reaching this part.

---

I see you already moved the selectedReports to the build function anyways. I just don't get the bug you're mentioning.

---

The error is swallowed [by the Add function](https://github.com/smartcontractkit/chainlink-ccip/blob/3ba6bac6203a27189da9c2ecc82b83e0415a17bc/execute/report/builder.go#L143) to allow continuing with other reports in case there is enough space for a smaller message. So we end up hitting this line even when the report isn't really selected.